requester.go

```diff
@@ -0,0 +1,352 @@
+	TransferEncoding []string
+	// Close indicates whether the connection should be closed after the request
+	Close bool
+	// Host is the host to use for the request
+	Host string
+	// Trailer is the trailer for the request
+	Trailer http.Header
+	// QueryParams are added to the request, in addition to any query params already encoded in the URL
+	QueryParams url.Values
+	// Body can be set to a string, []byte, io.Reader, or a struct; if set to a string, []byte, or io.Reader, the value will be used as the body of the request
+	// If set to a struct, the Marshaler will be used to marshal the value into the request body
+	Body interface{}
+	// Marshaler will be used to marshal the Body value into the body of the request.
+	Marshaler Marshaler
+	// Doer holds the HTTP client for used to execute httpsling
+	Doer Doer
+	// Middleware wraps the Doer
+	Middleware []Middleware
+	// Unmarshaler will be used by the Receive methods to unmarshal the response body
+	Unmarshaler Unmarshaler
+}
+
+// New returns a new Requester, applying all options
+func New(options ...Option) (*Requester, error) {
+	b := &Requester{}
+	err := b.Apply(options...)
+
+	if err != nil {
+		return nil, merry.Wrap(err)
+	}
+
+	return b, nil
+}
+
+// MustNew creates a new Requester, applying all options
+func MustNew(options ...Option) *Requester {
+	b := &Requester{}
+	b.MustApply(options...)
+
+	return b
+}
+
+func cloneURL(url *url.URL) *url.URL {
+	if url == nil {
+		return nil
+	}
+
+	urlCopy := *url
+
+	return &urlCopy
+}
+
+func cloneValues(v url.Values) url.Values {
+	if v == nil {
+		return nil
+	}
+
+	v2 := make(url.Values, len(v))
+
+	for key, value := range v {
+		v2[key] = value
+	}
+
+	return v2
+}
+
+func cloneHeader(h http.Header) http.Header {
+	if h == nil {
+		return nil
+	}
+
+	h2 := make(http.Header)
+
+	for key, value := range h {
+		h2[key] = value
+	}
+
+	return h2
+}
+
+// Clone returns a deep copy of a Requester
+func (r *Requester) Clone() *Requester {
+	s2 := *r
+	s2.Header = cloneHeader(r.Header)
+	s2.Trailer = cloneHeader(r.Trailer)
+	s2.URL = cloneURL(r.URL)
+	s2.QueryParams = cloneValues(r.QueryParams)
+
+	return &s2
+}
+
+// Request returns a new http.Request
+func (r *Requester) Request(opts ...Option) (*http.Request, error) {
+	return r.RequestContext(context.Background(), opts...)
+}
+
+// RequestContext does the same as Request, but requires a context
+func (r *Requester) RequestContext(ctx context.Context, opts ...Option) (*http.Request, error) {
+	reqs, err := r.withOpts(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	bodyData, ct, err := reqs.getRequestBody()
+	if err != nil {
+		return nil, err
+	}
+
+	urlS := ""
+	if reqs.URL != nil {
+		urlS = reqs.URL.String()
+	}
+
+	req, err := http.NewRequest(reqs.Method, urlS, bodyData)
+	if err != nil {
+		return nil, merry.Prepend(err, "creating request")
+	}
+
+	// if we marshaled the body, use our content type
+	if ct != "" {
+		req.Header.Set("Content-Type", ct)
```

Comment:
You already have a defined constant for this
```suggestion
		req.Header.Set(HeaderContentType, ct)
```